### ISSUE TITLE
Windows 10 20H2 Appears to Break SoapySDRServer --bind

### ISSUE THREAD

**Issue by aolszowka** [pothosware/SoapySDR#288]
## tl;dr;
In my setup after upgrading to Windows 10 20H2 `SoapySDRServer.exe --bind` no longer appears to work (some firewall corner case is filtering this). I am working around this by explicitly declaring the interface to bind to like so: `SoapySDRServer.exe --bind="0.0.0.0"`

## The Long
After experiencing a switch outage, I power cycled all my equipment, sadly this means I took the Windows 10 20H2 update on the Windows box I am using to host several Nooelec NESDR SMArt v4 devices. After the machine came back on I was no longer able to use CubicSDR to "see" the devices hosted on the Windows Machine.

After several hours of debugging and updating packages (I am now on the latest Windows `PothosSDR-2020.12.28-vc14-x64`) I believe that there is an issue with Windows and the way that `SoapySDRServer.exe --bind` behave.

Here is a truncated version that gets to the jist of this:

1. Ran `SoapySDRServer.exe --bind`
2. On my RaspberryPi I ran `nmap 10.1.0.10 -Pn -p 55132` which returned the following:
```
Host discovery disabled (-Pn). All addresses will be marked 'up' and scan times will be slower.
Starting Nmap 7.91 ( https://nmap.org ) at 2021-01-02 21:07 UTC
Nmap scan report for xxxxx.localdomain (10.1.0.10)
Host is up.

PORT      STATE    SERVICE
55132/tcp filtered unknown
```
3. What?! `filtered` - This would seem to indicate some type of Firewall Issue, but the Firewall should have already been configured from before the Windows update to allow SoapySDRServer through.
4. Disabled Windows Firewall.
5. Reran `nmap 10.1.0.10 -Pn -p 55132` on the RPi
```
Host discovery disabled (-Pn). All addresses will be marked 'up' and scan times will be slower.
Starting Nmap 7.91 ( https://nmap.org ) at 2021-01-02 21:21 UTC
Nmap scan report for xxxxx.localdomain (10.1.0.10)
Host is up (0.00073s latency).

PORT      STATE  SERVICE
55132/tcp closed unknown
```
6. WTF? `closed`? This makes zero sense!
7. Kill SoapySDRServer, relaunch with `SoapySDRServer.exe --bind="10.1.0.10"`
8. Reran `nmap 10.1.0.10 -Pn -p 55132` on the RPi
```
Host discovery disabled (-Pn). All addresses will be marked 'up' and scan times will be slower.
Starting Nmap 7.91 ( https://nmap.org ) at 2021-01-02 21:23 UTC
Nmap scan report for xxxxx.localdomain (10.1.0.10)
Host is up (0.00090s latency).

PORT      STATE SERVICE
55132/tcp open  unknown
```
9. Yeah that looks way better, lets turn the firewall back on
10. Firewall back on, Reran `nmap 10.1.0.10 -Pn -p 55132`
```
Host discovery disabled (-Pn). All addresses will be marked 'up' and scan times will be slower.
Starting Nmap 7.91 ( https://nmap.org ) at 2021-01-02 21:26 UTC
Nmap scan report for xxxxx.localdomain (10.1.0.10)
Host is up (0.00082s latency).

PORT      STATE SERVICE
55132/tcp open  unknown
```
11. Yeah that looks as I would have expected, the Firewall was previously configured correctly so what gives?
12. Lets try messing with some of the binding arguments (unsure if this is expected to work): `SoapySDRServer.exe --bind="0.0.0.0"`
13. Reran `nmap 10.1.0.10 -Pn -p 55132` on the RPi
```
Host discovery disabled (-Pn). All addresses will be marked 'up' and scan times will be slower.
Starting Nmap 7.91 ( https://nmap.org ) at 2021-01-02 21:28 UTC
Nmap scan report for xxxxx.localdomain (10.1.0.10)
Host is up (0.00090s latency).

PORT      STATE SERVICE
55132/tcp open  unknown
```
14. Yeah that is legit, lets try ONE MORE TIME `SoapySDRServer.exe --bind`
15. Reran `nmap 10.1.0.10 -Pn -p 55132` on the RPi
```
Host discovery disabled (-Pn). All addresses will be marked 'up' and scan times will be slower.
Starting Nmap 7.91 ( https://nmap.org ) at 2021-01-02 21:29 UTC
Nmap scan report for xxxxx.localdomain (10.1.0.10)
Host is up.

PORT      STATE    SERVICE
55132/tcp filtered unknown
```
16. DUDE WHAT GIVES? My only idea is that this is something to do with IPv6, maybe there is some corner case in Windows Defender Firewall that is getting tripped out now? Whatever I'll just use the `--bind="0.0.0.0"` work around.
17. Updates the Scheduled Task that runs this in the background on system startup with highest privilages without requiring user login.
18. CubicSDR doesn't automagically detect this instance, I think it used to? (**REEEEEEEEEE**):
![image](https://user-images.githubusercontent.com/2592881/103467419-24f64e00-4d0c-11eb-8a80-5f9efdfd3e37.png)
19. Reran `nmap 10.1.0.10 -Pn -p 55132` on the RPi
```
Host discovery disabled (-Pn). All addresses will be marked 'up' and scan times will be slower.
Starting Nmap 7.91 ( https://nmap.org ) at 2021-01-02 21:53 UTC
Nmap scan report for xxxxx.localdomain (10.1.0.10)
Host is up (0.00083s latency).

PORT      STATE SERVICE
55132/tcp open  unknown
```
20. Well it is open, if you connect to it manually, It works, but that stinks, but at least it is working now:
![image](https://user-images.githubusercontent.com/2592881/103467428-39d2e180-4d0c-11eb-9b36-e53acb94f54a.png)

I stopped digging after this, but if it would help to dig more let me know and I am willing to help to have someone else avoid this headache.

Thank you

**Comment by guruofquality** [comment]
The discovery operates over a multi-cast protocol SSDP: https://en.wikipedia.org/wiki/Simple_Service_Discovery_Protocol

On the client side, if you set the environment variable SOAPY_SDR_LOG_LEVEL to DEBUG, you should see prints about discovered servers.

It may also be worth watching wireshark with the filter for SSDP to see if its being blocked in or maybe out.

Both the client side of SoapyRemote should be requesting network servers to send, and on the server side, the server should be regularly sending out discovery info as well as replying to those requests.
